### PR TITLE
Feature/new schema

### PR DESCRIPTION
**Pull request by BenjaminStuermer**
Refactored the faceted search to support the new schema

### CODE REVIEW

File: application/src/PacificaSearchBundle/Controller/RestController.php [hunk 1]
```diff
@@ -76,48 +47,43 @@ public function postFilterPageAction($type, $pageNumber, Request $request) : Res
...
         foreach ($this->getFilterableRepositories() as $type => $repository) {
-            $filterPages[$type] = $repository->getFilteredPage(
-                $filter,
-                $this->getPageByType($type)
+            // Make a copy of the set of transactions but remove each type's own filter results - the contents of any
+            // given filter type are not constrained by what's already been chosen in that filter type, only by the
+            // selection of other filter types. For example, if you pick a user, you can still pick any other user, not
+            // only other users that share transactions with the original user.
+            $transIds = $transactionIdsByFilterItem;
+            unset($transIds[$type]);
```
Comment (AmericanUmlaut) [inline application/src/PacificaSearchBundle/Controller/RestController.php]: I think this must be a bug. Array assignment is by reference, so I'm unsetting $transactionIdsByFilterItem's value, too. Fix this.

File: application/web/js/namespace.js [hunk 2]
```diff
@@ -6,5 +6,11 @@
         throw new Error('PacificaSearch is already defined, did you include this file twice?');
     }
 
-    window.PacificaSearch = {};
+    window.PacificaSearch = {
+        /**
+         * How many items to load initially into each faceted search type, and how many to load on each click of the
+         * "more" link.
+         */
+        FACETED_SEARCH_ITEMS_PER_PAGE : 3
```
Comment (AmericanUmlaut) [inline application/web/js/namespace.js]: Double-check that this is being used. Aren't we just passing the page number and not the size? Either way this is redundant with the PHP constant, so one of the two should probably be removed

File: application/web/js/pacifica_search.js [hunk 3]
```diff
@@ -5,7 +5,7 @@
     var DomMgr = PacificaSearch.DomManager;
 
     $(function () {
-        $$('#search_filter')
+        $$('#search_filter  ')
```
Comment (BenjaminStuermer) [inline application/web/js/pacifica_search.js]: Eh?

File: application/web/js/utilities.js [hunk 4]
```diff
@@ -36,6 +36,22 @@
 
             return attrValue;
         },
+        /**
+         * Generate an error if the passed value is not an integer
+         * @param val Note any number of arguments can be passed, all will be validated
+         */
+        assertInteger : function (val) {
```
Comment (BenjaminStuermer) [inline application/web/js/utilities.js:43]: We should move to compiling our JS so this kind of thing can just be a regular variadic function

**Pull request merged**